test/wizards/simple.js

```diff
@@ -181,6 +204,283 @@ module.exports = {
     type: 'form'
   },
   tests: {
+    'Test Wizard valid submission': (form, done) => {
+      // Check current page
+      assert.equal(form.page, 0);
+      Harness.testElements(form, 'input[type="text"]', 2);
+      let buttonsToValid = ['Cancel', 'Next'];
+      let buttonsText =  _map(Harness.testElements(form, 'button'), (button) => button.innerText);
+      assert.deepEqual(buttonsText, buttonsToValid);
+
+      // Try to go to previous page. Stay on page 0
+      return Harness.testWizardPrevPage(form)
+        .then(() => {
+          // Should returns a reject
+          done('Should not be called');
+        })
+        .catch((error) => {
+          // Catch prev page error
+          assert.equal(form.page, 0);
+          assert.equal(error, 'Page not found');
+        })
+
+
+        .then(() => {
+          // Try to submit
+          return form.submit()
+        })
+        .then(() => {
+          // Should returns a reject
+          done('Should not be called');
+        })
+        .catch((error) => {
+          // Should returns a reject
+          assert.equal(error, 'Invalid Submission');
+        })
+
+
+        .then(() => {
+          // Try to go to next page. Stay on page 0
+          return Harness.testWizardNextPage(form, [{
+            component: 'a',
+            message: 'a is required'
+          }])
+        })
+        .then(() => {
+          // Should returns a reject
+          done('Should not be called');
+        })
+        .catch((error) => {
+          // Catch next page error
+          assert.equal(form.page, 0);
+        })
+
+
+        .then(() => {
+          // Write data
+          let wizardData = {
+            a: 'a',
+            b: 'b'
+          };
+          Harness.testSubmission(form, {
+            data: wizardData
+          });
+          assert.equal(form.page, 0);
+        })
+        .then(() => {
+          // Go to next page.
+          return Harness.testWizardNextPage(form);
+        })
+        .catch((error) => {
+          // Should returns a success
+          done(error);
+        })
+        .then(() => {
+          // Check next page
+          assert.equal(form.page, 1);
+          Harness.testElements(form, 'input[type="text"]', 2);
+          let buttonsToValid = ['Cancel', 'Previous', 'Next'];
+          let buttonsText =  _map(Harness.testElements(form, 'button'), (button) => button.innerText);
+          assert.deepEqual(buttonsText, buttonsToValid);
+        })
+
+
+        .then(() => {
+          // Click on cancel.
+          return form.cancel();
+        })
+        .catch((error) => {
+          // Should returns a success
+          done(error);
+        })
+        .then(() => {
+          // Check start page
+          assert.deepEqual(form.data, {
+            a: '',
+            b: '',
+            c: '',
+            d: ''
+          });
```

Comment:
@travist I don't know if this a bug or not. But after cancel, I have to compare with an object containing empty string instead of an empty object.
I have the same issue at line https://github.com/formio/formio.js/pull/70/files/02c3a480e1f5bf0d4e7a0ff0db2a7cd7a22aca01#diff-2d594189a79b299c2731f90be3b29e25R349 .  I have to define `d` as an empty string